### PR TITLE
Update simple-get version to 4.0.1

### PR DESCRIPTION
**Pull request by smokhov**
This version fixes a vuln in the dependency.

- [x] Have you updated CHANGELOG.md?

### CODE REVIEW

**Review by DraftProducts — APPROVED** [review]

**Comment by LinusU** [comment]
Unfortunately this change is incompatible with our current Node.js version support:

https://github.com/Automattic/node-canvas/blob/0bccc05b384cc1f3e969cd9ed7859a95376ec88c/package.json#L65-L67

Would be very nice if `simple-get` could release a patch 3.x version that also fixes it 😅 

Asked for it upstream here: https://github.com/feross/simple-get/issues/74

**Comment by DraftProducts** [comment]
Why could we not upgrade minimal node engine ?
The current Node versions is Node17 🤔

**Comment by smokhov** [comment]
Yeah, Node versions <=10 are already EOL. The current LTS is 12 expiring this year...
- https://endoflife.date/nodejs

**Review by Koyamie — APPROVED** [review]

**Comment by smokhov** [comment]
If @LinusU 's objection is lifted and the PR is accepted; should I bump that node version (to >=12, >=14?) in my branch as a part of this PR or you guys would do it afterward?

**Comment by Narretz** [comment]
A bump in node requirement would need to be in a major release

**Comment by smokhov** [comment]
BTW, in your CI tests, you only run in Node 10+:

- https://github.com/Automattic/node-canvas/blob/master/.github/workflows/ci.yaml

have there been any tests done lately on <10?

**Comment by DraftProducts** [comment]
ℹ️ In the prebuild CI it goes to Node 8
- https://github.com/Automattic/node-canvas/blob/master/.github/workflows/prebuild.yaml#L27

**Comment by zbjornson** [comment]
Let's see if feross/simple-get#74 gets addressed quickly. Ideally users don't have to do a major version update to get a bugfix, but we can do that if necessary.

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ project adheres to [Semantic Versioning](http://semver.org/).
 * Stringify CanvasGradient, CanvasPattern and ImageData like browsers do. (#1639, #1646)
 * Add missing include for `toupper`.
 * Throw an error instead of crashing the process if `getImageData` or `putImageData` is called on a PDF or SVG canvas (#1853)
+* Bump `simple-get` to 4.0.1 to fix a depentency vulnerability
```
Comment (kayahr) [inline CHANGELOG.md:16]:
```suggestion
* Bump `simple-get` to 4.0.1 to fix a dependency vulnerability
```

Comment (smokhov) [reply CHANGELOG.md:16]: Yeah, I saw that sometime afterward... Was going to correct it when the decision is made which version to go with if simple-get were to release 3.1.1 or something... Thank you for thoroughness :)

**Comment by smokhov** [comment]
Someone made a PR for 2.x.x in simple-get in the meantime (https://github.com/feross/simple-get/pull/75)... Hopefully any of that gets some traction some time soon.

**Comment by smokhov** [comment]
Someone made a PR to merge the fix for 3.x.x: https://github.com/feross/simple-get/pull/76

**Comment by smokhov** [comment]
How likely canvas is to branch off? For example,

- 3.0.0: bumps the engine to 12 and applies the 4.0.1 fix (master)
- 2.9.1: branches off and waits for the backport of simple-get to 3.1.1; keeps the engine at 6

How receptive are the maintainers to this proposal?

**Comment by smokhov** [comment]
Yay, I emailed Feross and he replied! He gave access to @LinusU in his repo to commit the backports! :)

**Comment by LinusU** [comment]
I have published simple-get 3.1.1 now so this is no longer needed 🚀 

Thanks everyone involved in the discussion!

**Comment by miestasmia** [comment]
I still see simple-get ^3.03 with the vuln in package.json, where have you published an updated version of node-canvas?

**Comment by LinusU** [comment]
@miestasmia I have published version `3.1.1` and `2.8.2` to npm.

Unfortunately I don't know how to get in contact with whomever manages the security advisories...

**Comment by miestasmia** [comment]
@LinusU I see no 2.8.2 on GitHub or npm. I get a simple missing version error when trying to install 2.8.2.

**Comment by Narretz** [comment]
I think @LinusU meant simple-get versions

**Comment by smokhov** [comment]
@LinusU -- this is prematurely closed, canvas still refers to 3.0.3 instead of 3.1.1:

https://github.com/Automattic/node-canvas/blob/0bccc05b384cc1f3e969cd9ed7859a95376ec88c/package.json#L53

**Comment by wizevlad** [comment]
Even tho 3.1.1 is backported, `npm audit` still wants `simple-get@4.0.1` minimum. Can you release a new version of canvas that uses 4.0.1 please? cc @LinusU

**Comment by Narretz** [comment]
@LinusU shouldn't make any hasty decisions. simple-get 4.0.1 would mean to drop support for node 6. Which is probably okay, but needs a major version bump.

Also. There hasn't even been a release of canvas which bumps the minimum simple-get dependency to 3.1.

And: in the meantime, resolutions should work to fix the audits? [yarn1](https://classic.yarnpkg.com/lang/en/docs/selective-version-resolutions/) , [yarn2+](https://yarnpkg.com/configuration/manifest/#resolutions), [npm8+](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides)

**Comment by zbjornson** [comment]
simple-get 3.1.1 (the fixed version) is in range of ^3.0.3 (the version spec'ed by node-canvas). We can publish a node-canvas 2.9.1, but making prebuilds are a slow pain, so until then, you just need to update simple-get. If you're using yarn, you can delete `simple-get` from your yarn.lock and re-run `yarn`.

> Can you release a new version of canvas that uses 4.0.1 please

This has to be in node-canvas 3.x (major bump).

Any help is appreciated getting the vulnerability advisories updated to acknowledge that simple-get 3.1.1 and 2.8.2 are not vulnerable.

**Comment by smokhov** [comment]
> simple-get 3.1.1 (the fixed version) is in range of ^3.0.3 (the version spec'ed by node-canvas). We can publish a node-canvas 2.9.1, but making prebuilds are a slow pain, so until then, you just need to update simple-get. If you're using yarn, you can delete `simple-get` from your yarn.lock and re-run `yarn`.

OK, then. Perhaps close #1977 then as well, thank you.

> > Can you release a new version of canvas that uses 4.0.1 please
> 
> This has to be in node-canvas 3.x (major bump).

Yes, please :)
 
> Any help is appreciated getting the vulnerability advisories updated to acknowledge that simple-get 3.1.1 and 2.8.2 are not vulnerable. _edit_ I've notified Snyk and emailed GitHub.

For my future ref who is 'Snyk'? :)

**Comment by zbjornson** [comment]
The GitHub vulnerability database is updated: https://github.com/advisories/GHSA-wpg7-2c88-r8xv
Snyk is another vulnerability database and has not yet updated: https://security.snyk.io/vuln/SNYK-JS-SIMPLEGET-2361683

**Pull request closed without merging**